ambry-frontend/src/main/java/com/github/ambry/frontend/PostAccountsHandler.java

```diff
@@ -135,11 +141,57 @@ private void start() {
      */
     private Callback<Long> fetchAccountUpdateBodyCallback(RetainingAsyncWritableChannel channel) {
       return buildCallback(frontendMetrics.postAccountsReadRequestMetrics, bytesRead -> {
-        updateAccounts(readJsonFromChannel(channel));
+        JSONObject jsonPayload = readJsonFromChannel(channel);
+        ReadableStreamChannel outputChannel;
+        if (restRequest.getPath().startsWith(Operations.UPDATE_ACCOUNT_CONTAINERS)) {
+          logger.debug("Got request for {} with payload {}", Operations.UPDATE_ACCOUNT_CONTAINERS, jsonPayload);
+          JSONObject outputPayload = updateContainers(jsonPayload);
+          outputChannel = serializeJsonToChannel(outputPayload);
+        } else {
+          updateAccounts(jsonPayload);
+          outputChannel = new ByteBufferReadableStreamChannel(ByteBuffer.allocate(0));
+        }
         restResponseChannel.setHeader(RestUtils.Headers.DATE, new GregorianCalendar().getTime());
-        restResponseChannel.setHeader(RestUtils.Headers.CONTENT_LENGTH, 0);
-        finalCallback.onCompletion(null, null);
-      }, uri, LOGGER, finalCallback);
+        restResponseChannel.setHeader(RestUtils.Headers.CONTENT_TYPE, RestUtils.JSON_CONTENT_TYPE);
+        restResponseChannel.setHeader(RestUtils.Headers.CONTENT_LENGTH, outputChannel.getSize());
+        finalCallback.onCompletion(outputChannel, null);
+      }, uri, logger, finalCallback);
+    }
+
+    /**
+     * Process the request json and call {@link AccountService#updateContainers} to add or update containers.
+     * @param containersPayload the request json containing the containers to update.
+     * @throws RestServiceException
+     */
+    private JSONObject updateContainers(JSONObject containersPayload) throws RestServiceException {
```

Comment:
hmm, I really should have put this blocking work in a thread pool instead of running it directly in the preceding callback. I guess we can do that at in the future.